lib/spack/spack/version/version_types.py

```diff
@@ -1019,6 +1024,9 @@ def __hash__(self):
         return hash(tuple(self.versions))
 
     def __str__(self):
+        if not self.versions:
+            return ""
```

Comment:
Why is this one also necessary? Looking at the class, versions could be an empty list - and the expression below returns `""` in that case. Am I missing some corner case?

Reply:
This is another fast path for the common case (like the one for variants)

Reply:
:thinking: I don't understand how this is a fast path, but it doesn't harm.